cpp/src/io/orc/orc_gpu.hpp

```diff
@@ -102,21 +102,22 @@ struct DictionaryEntry {
 struct ColumnDesc {
   uint8_t const* streams[CI_NUM_STREAMS];  // ptr to data stream index
   uint32_t strm_id[CI_NUM_STREAMS];        // stream ids
-  uint32_t strm_len[CI_NUM_STREAMS];       // stream length
+  uint32_t strm_len[CI_NUM_STREAMS];       // stream length TODO check for 64
   uint32_t* valid_map_base;                // base pointer of valid bit map for this column
   void* column_data_base;                  // base pointer of column data
-  uint32_t start_row;                      // starting row of the stripe
-  uint32_t num_rows;                       // number of rows in stripe
-  uint32_t column_num_rows;                // number of rows in whole column
-  uint32_t num_child_rows;                 // store number of child rows if it's list column
-  uint32_t num_rowgroups;                  // number of rowgroups in the chunk
-  uint32_t dictionary_start;               // start position in global dictionary
-  uint32_t dict_len;                       // length of local dictionary
-  uint32_t null_count;                     // number of null values in this stripe's column
-  uint32_t skip_count;                     // number of non-null values to skip
-  uint32_t rowgroup_id;                    // row group position
-  ColumnEncodingKind encoding_kind;        // column encoding kind
-  TypeKind type_kind;                      // column data type
+  uint64_t start_row;                      // starting row of the stripe
+  uint64_t num_rows;                       // number of rows in stripe
+  uint64_t column_num_rows;                // number of rows in whole column
+  uint32_t num_child_rows;  // store number of child rows if it's list column (output, 32bit should
+                            // be enough)
+  uint32_t num_rowgroups;   // number of rowgroups in the chunk
+  uint32_t dictionary_start;         // start position in global dictionary
```

Comment:
Can global dict be very large thus we need 64bit?

Reply:
Yup.
Took me a while to figure out what a "global dictionary" is. Looks like it's all dicts smushed together, and each one can have up to max<int32_t> elements.